frontend/src/components/SupportedAppTitle.tsx

```diff
@@ -10,18 +10,34 @@ type SupportedAppTitleProps = {
 };
 
 const SupportedAppTitle: React.FC<SupportedAppTitleProps> = ({ odhApp, showProvider = false }) => {
+  const { title, icon } = React.useMemo(() => {
```

Comment:
Not sure this memo is needed -- everything is deterministic.

Can you help me understand why you think you need/want to use memo?

Reply:
I am just used to using memo on things that work with arrays to squeeze out performance. i.e. the slicing and isRedHatSupported functions. I can remove the memo. Is it better to include this as an inline function in the JSX or as a separate constant

Reply:
> Is it better to include this as an inline function in the JSX or as a separate constant

I personally would do it just flat in the render function -- but if you want to build a splitter, have at it. Inline functions are fine when they are hyper specific to the implementation of the component; just be mindful of over-polluting the render scope, making it harder to read what's going on.

Reply:
how it would be as a constant

``` jsx
const { title, icon } = (() => {
    const split = odhApp.spec.displayName?.split(' ');

    if (isRedHatSupported(odhApp)) {
      return {
        title: `${split.slice(0, -1).join(' ')} `,
        icon: (
          <span style={{ whiteSpace: 'nowrap' }}>
            <span style={{ verticalAlign: 'text-bottom' }}>{split.pop()}</span>
            <Tooltip removeFindDomNode content={`${ODH_PRODUCT_NAME} certified and supported`}>
              <img
                style={{ marginLeft: 'var(--pf-global--spacer--xs)' }}
                src="../images/CheckStar.svg"
                alt={`${ODH_PRODUCT_NAME} certified and supported`}
              />
            </Tooltip>
          </span>
        ),
      };
    } return { title: odhApp.spec.displayName, icon: undefined };
  })()
```

Reply:
Hmm -- not sure you need to do that... I'm not _against_ IIFEs -- I just don't see a lot of the need for them. You could make this a function outside of the component (or even as a function in the component) just as easily (the dependency is just the odhApp).